tests/integration-tests-utils/src/main/java/org/apache/bookkeeper/tests/MavenClassLoader.java

```diff
@@ -102,6 +106,47 @@ public static MavenClassLoader forBookKeeperVersion(String version) throws Excep
         return forArtifact("org.apache.bookkeeper:bookkeeper-server:" +  version);
     }
 
+    public Object getClass(String className) throws Exception {
+        return Class.forName(className, true, classloader);
+    }
+
+    public Object callStaticMethod(String className, String methodName, ArrayList<?> args) throws Exception {
+        Class<?> klass = Class.forName(className, true, classloader);
+
+        try {
+            Class<?>[] paramTypes = args.stream().map((a)-> a.getClass()).toArray(Class[]::new);
+            return klass.getMethod(methodName, paramTypes).invoke(null, args.stream().toArray(Object[]::new));
+        } catch (NoSuchMethodException nsme) {
+            // maybe the params are primitives
+            Class<?>[] paramTypes = args.stream().map((a) -> {
```

Comment:
I never used such kind of code for coercions. Do you this it is needed?
I will look into my code and send some pointer

Reply:
It is. int.class is different to Integer.class, so without this, it won't find the method. I hit it when trying to find the static constructor for BKException. Groovy passed in an Integer as the argument. The method itself takes an int.

Reply:
Sure, I know, my point was about this way of doing coercions which I never seen and I wrote lot of code about this problems.
Any way this works and I think we can keep your proposal. I have no time today and this weekend and no access to laptop.

Reply:
There's probably a much better way to do this, I just went for the quickest that works.